### DISCUSSION TITLE
Unify gene names when cross species integration analysis

### DISCUSSION THREAD

**Discussion by whongfeiHK** [satijalab/seurat#5259] [Q&A]
Dear all,

I am doing cross species integration sc analysis between human and rat. However I found gene naming rules is different between them. For example, SNAI2 in human sample but Snai2 in rat, and also the '_' and '-', and may other differnet rules. I think this may cased by different refdata.gtf in cellranger. There are seems only one gtf named R_nor6.0 for Rat.

Before rename in R directly, may I ask you some better suggestions for that? I hope some of you may have handled this problem successfully. Many thanks.

Dale

**Comment by samuel-marsh** [comment]
Hi Dale,

Yes for analysis across species you will need to rename one of the matrcies.  But importantly you don't want to just rename things you want to refine your analysis to only include homologs/orthologs and rename them appropriately.  For instance CD33 in human and Cd33 in mouse have the same name but are not homologous genes so purely renaming will not fix that issue.  The bioMart package can be used to align and detect homologs/orthologs across species.  I would suggest also checking out many published works that have done this type of analysis for examples of how they have done this.

Best,
Sam
